arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -73,6 +83,60 @@ final class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdate
      */
     private var arCoreInstallRequested: Boolean = false
 
+    /**
+     * A background task used to poll ArCoreApk to set the value of ARCore availability for the current device.
+     *
+     * The ArCoreApk.getInstance().checkAvailability() function may inititate a query to a remote service to determine compatibility, in which case
+     * it immediately returns ArCoreApk.Availability.UNKNOWN_CHECKING. This leaves us unable to determine if the device
+     * is compatible with ARCore until the value is retrieved. See: https://developers.google.com/ar/reference/java/arcore/reference/com/google/ar/core/ArCoreApk#checkAvailability(android.content.Context)
+     *
+     * We should not be calling ArCoreApk.getInstance().requestInstall() until we've received one of the SUPPORTED_...
+     * values so this job allows us to do this. See: https://developers.google.com/ar/reference/java/arcore/reference/com/google/ar/core/ArCoreApk#requestInstall(android.app.Activity,%20boolean)
+     *
+     * @since 100.6.0
+     */
+    private val checkArCoreJob: Job by lazy {
+        GlobalScope.launch {
+            while (ArCoreApk.getInstance().checkAvailability(context) == ArCoreApk.Availability.UNKNOWN_CHECKING) {
+                Thread.sleep(100)
+            }
+            arCoreAvailability = ArCoreApk.getInstance().checkAvailability(context)
+        }
+    }
+
+    /**
+     * This property calls the observable function set on it during the setting of the value. If necessary, the
+     * observable is used in tandem with [checkArCoreJob] due to the synchronous nature of the
+     * ArCoreApk.getInstance().checkAvailability() function. If we know that the device is compatible with ARCore but
+     * ARCore isn't currently installed, or the version is older than the version used in this library, we request an
+     * installation of ARCore using the current context. If ARCore is already installed, we set [isUsingARCore] to true.
+     *
+     * @since 100.6.0
+     */
+    private var arCoreAvailability: ArCoreApk.Availability by Delegates.observable(ArCoreApk.Availability.SUPPORTED_INSTALLED) { _, _, newValue ->
```

Comment:
Then the coroutine would always fire.

Reply:
Does setting the initial property value trigger the delegate? If that's the case, then it would currently set `isUsingARCore` to `YES`, before we even have determined if ARCore is installed...

Reply:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.properties/-delegates/observable.html not according to the documentation. It should only fire when it changes.

Reply:
Let's discuss in a call, along with the initialisation sequence.

Reply:
Having read back on this, the coroutine wouldn't always fire. I'll changed it to `UNKNOWN_CHECKING`